runtime/compiler/spotbugs.vim

```diff
@@ -0,0 +1,119 @@
+" Vim compiler file
+" Compiler:     Spotbugs (Java static checker; needs javac compiled classes)
+" Maintainer:   @konfekt and @zzzxywvut
+" Last Change:  2024 nov 12
+
+if exists("current_compiler") | finish | endif
+
+let s:cpo_save = &cpo
+set cpo&vim
+
+" Unfortunately Spotbugs does not output absolute paths, so you need to
+" pass the directory of the files being checked as `-sourcepath` parameter.
+" The regex, auxpath and glob try to include all dependent classes of the
+" current buffer. See https://github.com/spotbugs/spotbugs/issues/856
+
+" FIXME: When "search()" is used with the "e" flag, it makes no _further_
+" progress after claiming an EOL match (i.e. "\_" or "\n", but not "$").
+" XXX: Omit anonymous class declarations
+let s:types = '\C\<\%(\.\@1<!class\|@\=interface\|enum\|record\)\%(\s\|$\)'
+let s:names = '\C\<\%(\.\@1<!class\|@\=interface\|enum\|record\)\s*\(\K\k*\)\>'
+let s:slash = exists('+shellslash') && !&shellslash ? '\' : '/'
+
+if has('syntax') && exists('g:syntax_on') && exists('b:current_syntax') &&
+    \ b:current_syntax == 'java' && hlexists('javaClassDecl')
+
+  function! s:GetDeclaredTypeNames() abort
+    defer execute('keepjumps normal ``')
+    normal gg
+    let type_names = []
+    let lnum = search(s:types, 'eW')
+    while lnum > 0
+      if synIDattr(synID(lnum, (col('.') - 1), 0), 'name') == 'javaClassDecl'
+        let tokens = matchlist(getline(lnum)..getline(lnum + 1), s:names)
+        if !empty(tokens) | call add(type_names, tokens[1]) | endif
+      endif
+      let lnum = search(s:types, 'eW')
+    endwhile
+    return type_names
+  endfunction
+
+else
+  function! s:GetDeclaredTypeNames() abort
+    " Undo the unsetting of &hls, see below
+    if &hls | defer execute('set hls') | endif
+    " Possibly restore the current value for register "y", see below
+    defer execute('let @y = '..(!empty(string(@y)) ? string(@y) : '""'))
+    " Copy buffer contents for modification
+    silent %y y
+    new
+    defer execute('silent bwipeout')
```

Comment:
move this defer up to the other defers please, also use `bwipeout!`

Reply:
Hypothetically, if the process, say, to run out of memory  
between yanking the old buffer and allocating a new buffer,  
is it safe to wipe out a possibly changed _old_ buffer?

Reply:
good point. thanks

Reply:
Thank you, applied (without `!` after `bwipeout`)